teos/src/api/internal.rs

```diff
@@ -380,25 +424,156 @@ mod tests_private_api {
+            .await
+            .unwrap()
+            .into_inner();
+
+        assert!(matches!(response, msgs::GetAppointmentsResponse { .. }));
+    }
+
+    #[tokio::test]
+    async fn test_get_appointments_watcher() {
+        let internal_api = create_api().await;
+
+        // Add a user to send the appointments.
+        let (user_sk, user_pk) = get_random_keypair();
+        internal_api.watcher.register(UserId(user_pk)).unwrap();
+
+        // Generate two different appointments.
+        let appointment1 = generate_dummy_appointment(None).inner;
+        let appointment2 = generate_dummy_appointment(None).inner;
+        let locator1 = appointment1.locator.serialize();
+        let locator2 = appointment2.locator.serialize();
+
+        // Calculate their signature.
+        let signature1 = cryptography::sign(&appointment1.serialize(), &user_sk).unwrap();
+        let signature2 = cryptography::sign(&appointment2.serialize(), &user_sk).unwrap();
+
+        // Add them to the watcher.
+        internal_api
+            .watcher
+            .add_appointment(appointment1, signature1)
+            .unwrap();
         internal_api
             .watcher
-            .add_random_tracker_to_responder(UUID::new(appointment.locator, user_id));
+            .add_appointment(appointment2, signature2)
+            .unwrap();
 
+        // Query for each locator and assert it retrieves the correct appointment.
         let response = internal_api
-            .get_all_appointments(Request::new(()))
+            .get_appointments(Request::new(msgs::GetAppointmentsRequest {
+                locator: locator1.clone(),
+            }))
             .await
             .unwrap()
             .into_inner();
 
+        // The response should contain only one appointment with locator1.
         assert_eq!(response.appointments.len(), 1);
         assert!(matches!(
             response.appointments[0].appointment_data,
-            Some(msgs::appointment_data::AppointmentData::Tracker { .. })
+            Some(msgs::appointment_data::AppointmentData::Appointment(
+                msgs::Appointment {
+                    ref locator,
+                    ..
+                }
+            )) if *locator == locator1
+        ));
+
+        let response = internal_api
+            .get_appointments(Request::new(msgs::GetAppointmentsRequest {
+                locator: locator2.clone(),
+            }))
+            .await
+            .unwrap()
+            .into_inner();
+
+        // The response should contain only one appointment with locator2.
+        assert_eq!(response.appointments.len(), 1);
+        assert!(matches!(
+            response.appointments.first().unwrap().appointment_data,
+            Some(msgs::appointment_data::AppointmentData::Appointment(
+                msgs::Appointment {
+                    ref locator,
+                    ..
+                }
+            )) if *locator == locator2
+        ));
+    }
```

Comment:
This could be another loop comparing the response against the previously generated collection